script/parse-examples

```diff
@@ -3,28 +3,50 @@
 set -e
 
 cd "$(dirname "$0")/.."
+root="$PWD"
 
-function clone_repo {
-  owner=$1
-  name=$2
-  sha=$3
+clone_repo() {
+  local owner=$1
+  local name=$2
+  local sha=$3
+  local folder="$4"
+  local path="$root/examples/$name"
 
-  path=examples/$name
-  if [ ! -d "$path" ]; then
+  if [ -d "$path" ]; then
+    pushd "$path" > /dev/null
+    if [ "$(git rev-parse HEAD 2>/dev/null)" == "$sha"  ]; then
+      popd > /dev/null
+      return
+    else
+      popd > /dev/null
+      rm -rf "$path"
+      echo "Updating $owner/$name to $sha"
+    fi
+  else
     echo "Cloning $owner/$name"
-    git clone "https://github.com/$owner/$name" "$path"
   fi
 
+  mkdir -p "$path"
   pushd "$path" > /dev/null
-  if [ "$(git rev-parse head)" != "$sha"  ]; then
-    echo "Updating $owner/$name to $sha"
-    git fetch
-    git reset --hard $sha
+  git init
+  git remote add origin "https://github.com/$owner/$name"
+  git pull --ff-only --depth 1 origin "$sha"
+  if [ "${folder:-}" ]; then
+    while IFS= read -r file; do
+      echo "${folder}" "$file"
```

Comment:
This is a print debugging leftover. Not sure if you mind.

Reply:
Oh yeah, I missed that. What is this while loop for?

Reply:
It's for cleaning unrelated folders before the `**/*.ts` glob. In the TypeScript repository there's the [tests](https://github.com/microsoft/TypeScript/tree/master/tests) folder which has some broken files on purpose, so I thought I only wanted the `src` folder which is more certain to have only valid syntax.

This loop goes over the files in the directory and removes everything that's not `$folder` - in this case I was passing `src` as `$folder` since that's the only one I'm interested in.